src/thought/view-thoughts/thoughts-mind-map-view.js

```diff
@@ -1,281 +1,286 @@
+      this.createEmptyChild(targetThoughtId);
+    }
+  };
 
-      return thought;
-    });
-}
+  visNetworkHoldHandler = event => {
+    if (visNetworkHelper.clickedOnThought(event)) {
+      let targetThoughtId = visNetworkHelper.getTargetThoughtId(event);
+      console.info('=== Event === Node hold');
+      this.editThought(targetThoughtId);
+    }
+  };
 
-function thoughtClickHandler(targetThoughtId) {
-  // if clicking on the current thought, do nothing.
-  if (targetThoughtId === currentViewedThoughtId) {
-    console.info('Click was on the selected note, doing nothing')
-    return;
-  }
+  editThought(targetThoughtId) {
+    const thought = thoughtStorage.findThoughtById(targetThoughtId);
 
-  var currentViewedThoughtId_temp = _.find(brainVisNetwork.visNodes.getIds(),
-    function (nodeId) {
-      return nodeId == targetThoughtId;
+    if (thought.name === thoughtStorage.APP_FOLDER_NAME) {
+      console.info('It is not allowed to edit App root folder name');
+      return;
     }
-  );
 
-  console.debug('currentViewedThoughtId from visNodes: ', currentViewedThoughtId_temp);
+    if (!thought.isNote) {
+      console.info('It is not allowed to edit user-uploaded files');
+      return;
+    }
 
-  var node = brainVisNetwork.visNodes.get(currentViewedThoughtId_temp);
-  console.debug('node from visNodes: ', node);
+    this.setState({
+      currentViewedThought: thought,
+      editNote: {
+        id: targetThoughtId
+      }
+    });
+    brainVisNetwork.selectNote(targetThoughtId);
+  }
 
-  currentViewedThought = {
-    id: node.id,
-    name: node.label
+  onChangeClick = name => {
+    const {id} = this.state.editNote;
+    thoughtStorage.updateThoughtName({id, name})
+      .then(() => brainVisNetwork.updateNode({id, label: name}));
   };
 
-  console.debug('currentViewedThought from visNodes: ', currentViewedThought);
-
-  console.debug('targetThoughtId: ', targetThoughtId);
-  console.debug('brainVisNetwork.visNodes: ', brainVisNetwork.visNodes);
+  onDeleteClick = () => {
+    let note = this.state.currentViewedThought;
+    console.info('Deleting ' + note.name + '...');
+    thoughtStorage.remove(note)
+      .then(function () {
+        console.info('Deleted ' + note.name);
+        brainVisNetwork.deleteSelectedNode();
+      });
+  };
 
-  console.debug('brainVisNetwork.visNodes: ', brainVisNetwork.visNodes);
+  onUploadFileClick = () => {
+    let note = this.state.currentViewedThought;
+    console.info('[Event] Upload file click');
 
-  currentViewedThoughtId = targetThoughtId;
-  thoughtStorage.logTree();
-  var targetThought = thoughtStorage.findThoughtById(targetThoughtId);
+    window.open(thoughtStorage.getLinkToThought(this.state.currentViewedThought));
+  };
 
-  if (!targetThought) {
-    throw new Error('changeThought(): couldn\'t find targetThought in thoughtStorage by id: ', targetThoughtId);
+  closeThoughtNameEditor() {
+    this.setState({editNote: null});
```

Comment:
I think this should be something like `isNoteBeingEdited: boolean` or `noteMode: view | edit` or `isInNoteEditMode: boolean`.

Reply:
changed to "selectedNote", bacause it's object, not a boolean